packages/algob/src/internal/deployer.ts

```diff
@@ -945,6 +987,28 @@ export class DeployerRunMode extends DeployerBasicMode implements Deployer {
 		});
 	}
 
+	/**
+	 * Checkpoint feature disable on run mode
+	 * @param asaName name of asa
+	 * @param txConfirmation txn confirmation
+	 */
+	addCheckpointASA(asaName: string, txConfirmation: ConfirmedTxInfo): void {
+		throw new BuilderError(ERRORS.BUILTIN_TASKS.DEPLOYER_EDIT_OUTSIDE_DEPLOY, {
+			methodName: "addCheckpointASA",
+		});
+	}
```

Comment:
What if i am doing an `app call` in *RUN* mode (script), which is actually deploying a new app internally? In that case application deployment (via inner tx), would be *RUN* mode, not *DEPLOY* mode.

Reply:
I also think about it! I think we can run something like `yarn algob deploy ./script/run.js`. How you think about it?